content-src/components/Base/Base.jsx

```diff
@@ -114,17 +116,29 @@ export class _Base extends React.PureComponent {
     const {initialized} = App;
 
     const prefs = props.Prefs.values;
-    if ((prefs.asrouterExperimentEnabled || prefs.asrouterOnboardingCohort > 0) && window.location.hash === "#asrouter") {
+
+    // Enable asrouter content
+    if (prefs.asrouterExperimentEnabled && !asrouterContent.initialized) {
+      asrouterContent.init();
+    } else if (!prefs.asrouterExperimentEnabled && asrouterContent.initialized) {
+      asrouterContent.uninit();
+    }
+
+    if (prefs.asrouterExperimentEnabled && window.location.hash === "#asrouter") {
       return (<ASRouterAdmin />);
     }
 
     if (!props.isPrerendered && !initialized) {
       return null;
     }
-
     // Until we can delete the existing onboarding tour, just hide the onboarding button when users are in
     // the new simplified onboarding experiment. CSS hacks ftw
-    if (prefs.asrouterOnboardingCohort > 0) {
+    let isOnboardingEnabled = false;
+    try {
+      isOnboardingEnabled = JSON.parse(prefs["asrouter.messageProviders"]).find(i => i.id === "onboarding").enabled;
```

Comment:
could we avoid all this parsing by converting this pref to a `getPref` function?

Reply:
Also this could presumably false if onboarding is not defiled at all, no?

Reply:
`getPref` i think can only read booleans, integers, and strings, but we'd still have to parse the JSON value to get the `enabled` field, no?
And yes, it would stay false if it's not defined, which is i think is fine, because we'd want to fall back on the old onboarding experience in that case